request.el

```diff
@@ -977,15 +977,19 @@ removed from the buffer before it is shown to the parser function.
   (let* (;; Use pipe instead of pty.  Otherwise, curl process hangs.
          (process-connection-type nil)
          ;; Avoid starting program in non-existing directory.
-         (default-directory (expand-file-name "~/"))
+         (home-directory (if (tramp-tramp-file-p default-directory)
```

Comment:
Can `file-remote-p` be used instead `tramp-tramp-file-p` ? `tramp.el` is loaded when `tramp-tramp-file-p` is called. tramp.el is a big library, non-tramp users should not load it if it is needed.